npm/private/npm_package_store.bzl

```diff
@@ -151,6 +151,33 @@ If set, takes precendance over the package version in the NpmPackageInfo src.
     ),
 }
 
+def _fast_symlink(ctx, symlink_path, target_path, prefix):
+    symlink = ctx.actions.declare_symlink(symlink_path)
+    symlink_path = symlink.path
+
+    total = len(symlink_path)
+    for i in range(total):
+        if symlink_path[i] != target_path[i]:
+            for j in range(i, 0, -1):
+                if symlink_path[j] == "/":
+                    break
+            break
+    new_target_path = prefix + target_path[j:]
+
+    ctx.actions.symlink(
+        output = symlink,
+        target_path = new_target_path,
+    )
+    return symlink
+
+def _fast_join(path, *others):
+    result = path
+
+    for p in others:
+        result += "/" + p
```

Comment:
I think this would be slower then `path.join`? String concat is generally not recommended because it has to allocate a new string every time where `.format()` or `path.join` etc will do it optimally...

Reply:
We were previously calling `paths.join`; this code is a stripped-down [version of it](https://github.com/bazelbuild/bazel-skylib/blob/main/lib/paths.bzl#L70-L98). The key bit is to remove the call to `_is_absolute` since that is pretty slow.

I'm not sure what `path.join` is, but I can give `str.format` a shot, that's a good idea

Reply:
I pulled this out into https://github.com/aspect-build/rules_js/pull/1801